### PR TITLE
Handle guild data load errors better

### PR DESCRIPTION
**Pull request by Octol1ttle**
Previously, any errors in guild data load will cause the bot to be unusable in that guild. It didn't help that the end users had no information that something was wrong! Now, any errors will be logged better (with the full path to the file that couldn't be loaded), and the users will receive a message saying that functionality is degraded

The old way to save objects was to serialize them directly into streams opened by `File#Create`. This can cause problems if the serialization isn't completed, because `File#Create` overwrites the file with an empty one on the spot. Now, objects are first deserialized into a temporary file, then the original is replaced by the temporary, then the temporary is deleted.

Errors during guild data load would sometimes cause the bot to replace the corrupted file with a default one whenever a save is triggered. Now, guilds with load errors won't have their data saved to aid in debugging

### CODE REVIEW

File: locale/Messages.tt-ru.resx [hunk 1]
```diff
@@ -570,4 +570,10 @@
   <data name="MessagesClearedFiltered" xml:space="preserve">
       <value>вырезано {0} забавных сообщений от {1}</value>
   </data>
+  <data name="DataLoadFailedTitle" xml:space="preserve">
+      <value>произошёл пиздец в гилддате.</value>
```
Comment (mctaylors) [inline locale/Messages.tt-ru.resx:574]: not that rude!!!
```suggestion
      <value>произошёл глобальный разнос в гилддате.</value>
```

**Review by mctaylors — CHANGES_REQUESTED** [review]

**Comment by mctaylors** [comment]
Same question as [here](https://github.com/LabsDevelopment/Octobot/pull/171#issuecomment-1781168406).

**Comment by mctaylors** [comment]
> Same question as [here](https://github.com/LabsDevelopment/Octobot/pull/171#issuecomment-1781168406).

Please answer to this question first.

**Comment by Octol1ttle** [comment]
> sure

**Pull request merged**